lightningd/bitcoind.c

```diff
@@ -111,19 +111,19 @@ static struct io_plan *output_init(struct io_conn *conn, struct bitcoin_cli *bcl
 static void next_bcli(struct bitcoind *bitcoind, enum bitcoind_prio prio);
 
 /* For printing: simple string of args. */
-static char *bcli_args(const tal_t *ctx, struct bitcoin_cli *bcli)
+static char *bcli_args(const tal_t *ctx, const char **args)
```

Comment:
I would have instead renamed this to `bcli_args_direct`, then created a new `bcli_args`:

    static char *bcli_args(const tal_t *ctx, struct bitcoin_cli *bcli)
    {
        return bcli_args_direct(ctx, &bcli->args);
    }

This would have reduced the large number of changes later in the file.

A generally good rule is that we should avoid changing the interface of a function, instead change function name when we change interface and create a wrapper that retains previous interface.

Reply:
Thank you! Nice idea :)